src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs

```diff
@@ -0,0 +1,143 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.SqlVirtualMachine.Common
+{
+    /// <summary>
+    /// The base class for all Azure Sql virtual machne cmdlets.
+    /// </summary>
+    /// <typeparam name="M">Type of the model the cmdlet will be working on (e.g. AzureSqlVirtualMachineModel)</typeparam>
+    /// <typeparam name="A">Adapter used to call the REST APIs to perform actions on the specified model</typeparam>
+    public abstract class AzureSqlVirtualMachineCmdletBase<M, A> : AzureRMCmdlet
+    {
+        /// <summary>
+        /// Get the ResourceId property value of the model provided.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        protected virtual string GetResourceId(M model)
+        {
+            if (model != null && model is IEnumerable<object>)
```

Comment:
the "is" operator should return false if model is null so the first check "model !=null" may not be required. Please verify before removing if there are edge cases.